src/lib/y2network/sysconfig/interface_file.rb

```diff
@@ -35,22 +35,78 @@ class << self
 
         # Finds the ifcfg-* file for a given interface
         #
-        # @param name [String] Interface name
+        # @param interface [String] Interface name
         # @return [Sysconfig::InterfaceFile,nil] Sysconfig
-        def find(name)
-          return nil unless Yast::FileUtils.Exists(SYSCONFIG_NETWORK_DIR.join("ifcfg-#{name}").to_s)
-          new(name)
+        def find(interface)
+          return nil unless Yast::FileUtils.Exists(SYSCONFIG_NETWORK_DIR.join("ifcfg-#{interface}").to_s)
+          new(interface)
         end
 
-        def define_parameter(name, type = :string)
-          define_method name do
-            value = fetch(name.to_s.upcase)
+        # Defines a parameter
+        #
+        # This method adds a pair of methods to get and set the parameter's value.
+        #
+        # @param param_name [Symbol] Parameter name
+        # @param type       [Symbol] Type to be used (:string, :integer, :symbol, :ipaddr)
+        def define_parameter(param_name, type = :string)
+          params << param_name
+
+          define_method param_name do
+            return @values[param_name] if @values.key?(param_name)
+            value = fetch(param_name.to_s.upcase)
             send("value_as_#{type}", value)
           end
+
+          define_method "#{param_name}=" do |value|
+            @values[param_name] = value
```

Comment:
i would mention here in comment that write does not need convert, that write itself use always to_s for type. So all types have to have this method and also it have to return expected value. It is something I have to check and is not obvious from code.